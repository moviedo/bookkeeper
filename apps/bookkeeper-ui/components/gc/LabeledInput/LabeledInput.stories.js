export default {
  title: "grouped components / gc-labeled-input",
};

const Template = (_args, { argTypes }) => ({
  props: Object.keys(argTypes),
  template: `
  <gc-labeled-input v-bind="$props.$attrs">
    <template #label v-if="$props.labelSlot">
      {{$props.labelSlot}}
    </template>

    <template #hint v-if="$props.hintSlot">
      {{$props.hintSlot}}
    </template>
  </gc-labeled-input>`,
});

export const Default = Template.bind({});
Default.args = {
  $attrs: {
    id: "username",
    name: "username",
    placeholder: "Username",
  },
  labelSlot: "Username",
};

export const WithHint = Template.bind({});
WithHint.args = {
  ...Default.args,
  hintSlot: "Password is not available.",
};
