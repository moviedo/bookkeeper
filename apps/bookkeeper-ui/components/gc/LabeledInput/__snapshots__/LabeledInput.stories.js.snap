// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grouped components / gc-labeled-input Default smoke-test 2`] = `
<fieldset style="width: 320px;">
  <label class="text-base font-normal text-slate-600 pl-px"
         for="username"
  >
    Username
  </label>
  <input data-v-10e03f24
         type="text"
         class="default hover focus my-2"
         id="username"
         name="username"
         placeholder="Username"
  >
</fieldset>
`;

exports[`grouped components / gc-labeled-input WithHint smoke-test 2`] = `
<fieldset style="width: 320px;">
  <label class="text-base font-normal text-slate-600 pl-px"
         for="username"
  >
    Username
  </label>
  <input data-v-10e03f24
         type="text"
         class="default hover focus my-2"
         id="username"
         name="username"
         placeholder="Username"
  >
  <p data-v-34af387a
     class="text-sm font-normal pl-px error"
  >
    <svg data-v-34af387a
         xmlns="http://www.w3.org/2000/svg"
         viewbox="0 0 20 20"
         fill="currentColor"
         aria-hidden="true"
         class="inline h-5 w-5 pr-1"
    >
      <path data-v-34af387a
            fill-rule="evenodd"
            d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
            clip-rule="evenodd"
      >
      </path>
    </svg>
    Password is not available.
  </p>
</fieldset>
`;
